src/query_backend.h

```diff
@@ -1082,11 +1004,26 @@ class Sha1 {
     }
   }
 
-  /// \}
 
   Digest digest() {
     Digest d;
-    hash_.get_digest(d.val);
+    #if BOOST_VERSION_MINOR < 86
+        unsigned int tmp[5];
+    #else
+        unsigned char tmp[20];
+    #endif
```

Comment:
I'm not married to the variable name `block_size`, but how about this?
```suggestion
    unsigned int block_size = sizeof(unsigned int);
    #if BOOST_VERSION_MINOR < 86
        unsigned int tmp[CYCLUS_SHA1_NINT];
    #else
        unsigned char tmp[CYCLUS_SHA1_NINT * block_size];
    #endif
```